code/ship/ship.cpp

```diff
@@ -18985,3 +18986,48 @@ ship_subsys* ship_get_subsys_for_submodel(ship* shipp, int submodel)
 
 	return nullptr;
 }
+
+/**
+ * Is the requested type of ship arriving in the requested stage?
+ *
+ * Dock leaders & followers need to be handled separately such that
+ * leaders handle the warp process, but followers need to behave as though
+ * warping with respect to targeting, drawing thrusters, radar, AI
+ * targeting, etc
+ *
+ * @param stage Check one of stage1, stage2 or both
+ * @param dock_leader_only only the dock leader, or leader/followers/non-docked?
+ *
+ * @return is the ship arriving, bool
+ */
+bool ship::is_arriving(ship::warpstage stage, bool dock_leader_only)
+{
+	if (stage == ship::warpstage::BOTH) {
+		if (dock_leader_only == false) {
```

Comment:
If this is going to be a boolean, it should check the variable directly (with ! if necessary) not with a ==.